charts/kong/templates/_helpers.tpl

```diff
@@ -437,6 +437,8 @@ the template that it itself is using form the above sections.
   {{- $_ := set $autoEnv "KONG_PG_PORT" .Values.postgresql.service.port -}}
   {{- $pgPassword := include "secretkeyref" (dict "name" (include "kong.postgresql.fullname" .) "key" "postgresql-password") -}}
   {{- $_ := set $autoEnv "KONG_PG_PASSWORD" $pgPassword -}}
+{{- else if eq .Values.env.database "postgres" }}
+  {{- $_ := set $autoEnv "KONG_PG_PORT" "5432" }}
```

Comment:
What about KONG_PG_HOST in this case?

Reply:
Omitting `KONG_PG_HOST` should be fine, since we can expect that it's always defined in K8S deployments. `KONG_PG_PORT` relies on the default from kong.conf, which netcat has no way of knowing. The kong.conf default port is typically correct for most Postgres installs, but the default host is `localhost` and won't be valid unless Postgres is running within the same Pod as Kong.

That's not impossible to set up, but you'd have to rework a bunch of the chart to do it, so I'm comfortable saying that it shouldn't be necessary to explicitly add the default to autoEnv.

Reply:
Yeh, my thoughts were that if you use an external DB then you will add `pg_host` and expect it to default to Postgres port.

Besides, the `nc` check will hard fail if either is omitted.